src/protocols/secure_channel/CASESession.cpp

```diff
@@ -1038,11 +1057,19 @@ CHIP_ERROR CASESession::HandleSigma2(System::PacketBufferHandle && msg)
 
     // Validate responder identity located in msg_r2_encrypted
     // Constructing responder identity
-    SuccessOrExit(err = ValidatePeerIdentity(responderNOC, responderICAC, responderNodeId, responderPublicKey));
-
-    // Verify that responderNodeId (from responderNOC) matches one that was included
-    // in the computation of the Destination Identifier when generating Sigma1.
-    VerifyOrExit(mPeerNodeId == responderNodeId, err = CHIP_ERROR_INVALID_CASE_PARAMETER);
+    {
+        uint8_t rootCertBuf[kMaxCHIPCertLength];
+        MutableByteSpan fabricRCAC{ rootCertBuf };
+        CompressedFabricId unused;
+        FabricId responderFabricId;
+        SuccessOrExit(err = mFabricsTable->FetchRootCert(mFabricIndex, fabricRCAC));
+        SuccessOrExit(err = SetEffectiveTime());
+        SuccessOrExit(err = FabricTable::VerifyCredentials(responderNOC, responderICAC, fabricRCAC, mValidContext, unused, responderFabricId, responderNodeId, responderPublicKey));
```

Comment:
Why can't this use the API that takes fabric index instead of reimplementing it?

Reply:
I can change it, but it's going to have to then change back to this when we split up Sigma2 same as Sigma3.

Reply:
Done.